src/sysdolphin/baselib/aobj.c

```diff
@@ -259,45 +259,47 @@ void HSD_AObjFree(HSD_AObj* aobj)
 }
 
 void callbackForeachFunc(struct _HSD_AObj* aobj, void* obj, HSD_Type type,
-                         void (*func)(), AObj_Arg_Type arg_type,
-                         callbackArg* arg)
+                         void* func, AObj_Arg_Type arg_type, callbackArg* arg)
```

Comment:
This seems like a case where not specifying the argument types is cleaner. Requiring the explicit casts here adds no useful information

Reply:
A small price to pay imo.

Reply:
Just to clarify, `-Wstrict-prototypes` on clang is a warning but `-requireprotos` does not have any flexibility to it. MWCC will abort if it comes across non-compliant code with that option enabled. So I agree the casts suck but I don't think code like that outweighs the benefits.

Reply:
`void (*func)()` is compliant with -requireprotos though?

Reply:
I can double-check but I'm pretty sure it's not.

Reply:
Oh I get the issue, `void (*func)()` isn't C99 compliant so you're having trouble using it with clang-tools. That's probably gonna be an issue for a lot of modern tooling. I can deal with the ugliness in that case

Reply:
Actually no, I elevated the warning to an error in `compile_flags.txt` _because_ `-requireprotos` doesn't allow empty parentheses.

This was like a month ago though so I'm not 100% sure anymore.

Reply:
Well I just tested it and does compile with -requireprotos, but a stricter -Werror=strict-prototypes flag is probably a net improvement.

Reply:
I see, good to know, thanks.